python/ray/serve/_private/common.py

```diff
@@ -393,6 +394,7 @@ def __post_init__(self):
                 [
```

Comment:
Can we also fix the separator here? Using " " is problematic (if your node_id is empty, your new hash will be indistinguishable from old one)

Reply:
You can use a char that we know for sure wouldn't be used in either of these fields (":", ",", etc)

Reply:
Synced offline, will keep in this PR